google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -671,9 +701,18 @@ public long executeUpdate(Statement statement, UpdateOption... options) {
           throw new IllegalArgumentException(
               "DML response missing stats possibly due to non-DML statement as input");
         }
+        if (inlineCommitState == InlineCommitState.COMMITTING) {
+          if (resultSet.hasCommitResponse()) {
+            inlineCommitState = InlineCommitState.SUCCEEDED;
+            commitResponse = new CommitResponse(resultSet.getCommitResponse());
+          } else {
+            inlineCommitState = InlineCommitState.MISSING_COMMIT_RESPONSE;
+          }
+        }
         // For standard DML, using the exact row count.
         return resultSet.getStats().getRowCountExact();
       } catch (Throwable t) {
+        inlineCommitState = InlineCommitState.FAILED;
```

Comment:
You should only set this if autocommit was enabled.

Reply:
ok